tests/drivers/ethernet/eth_mac_config/src/main.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2025 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/ethernet.h>
+
+#include "eth_test_priv.h"
+
+ZTEST(ethernet_mac_config, test_eth_mac_local)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(eth_mac_local));
+	const struct vnd_ethernet_config *cfg = dev->config;
+	struct vnd_ethernet_data *data = dev->data;
+	const uint8_t addr[NET_ETH_ADDR_LEN] = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05};
+
+	zexpect_equal(data->mac_addr_load_result, 0);
+	zexpect_equal(cfg->mcfg.type, NET_ETH_MAC_STATIC);
+	zexpect_equal(cfg->mcfg.addr_len, NET_ETH_ADDR_LEN);
+	zexpect_mem_equal(cfg->mcfg.addr, addr, NET_ETH_ADDR_LEN);
+	zexpect_mem_equal(data->mac_addr, addr, NET_ETH_ADDR_LEN);
+}
+
+ZTEST(ethernet_mac_config, test_eth_mac_random)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(eth_mac_random));
+	const struct vnd_ethernet_config *cfg = dev->config;
+	struct vnd_ethernet_data *data = dev->data;
+
+	zexpect_equal(data->mac_addr_load_result, 0);
+	zexpect_equal(cfg->mcfg.type, NET_ETH_MAC_RANDOM);
+	zexpect_equal(cfg->mcfg.addr_len, 0);
+}
+
+ZTEST(ethernet_mac_config, test_eth_mac_random_prefix)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(eth_mac_random_prefix));
+	const struct vnd_ethernet_config *cfg = dev->config;
+	struct vnd_ethernet_data *data = dev->data;
+	uint8_t prefix[] = {0x00, 0x22, 0x44};
+
+	zexpect_equal(data->mac_addr_load_result, 0);
+	zexpect_equal(cfg->mcfg.type, NET_ETH_MAC_RANDOM);
+	zexpect_equal(cfg->mcfg.addr_len, 3);
+	zexpect_mem_equal(cfg->mcfg.addr, prefix, sizeof(prefix));
+
+	/* NOTE: Prefix with LAA bit set */
+	prefix[0] |= 0x02;
```

Comment:
Test Multicast/Broadcast bit cleared? (maybe default set it in `prefix`)

Reply:
I think it's a bit weird to set it in the overlay, as that would clearly be an unwanted value.